firmware/controllers/actuators/electronic_throttle.cpp

```diff
@@ -269,6 +274,29 @@ static percent_t currentEtbDuty;
 
 		etb1.dcMotor.Set(ETB_PERCENT_TO_DUTY(currentEtbDuty));
 
+	   	float etbError = absF(targetPosition - actualThrottlePosition); // Used for I-Term limiter and Error-detection
+
+
+        // ETB Deviation monitoring
+        if(etbError < 3 ){
+          start_time = chVTGetSystemTimeX(); //Updates timestamp as long as ETB-Error stays within 3% deviation
+        }
+      else {
+        }        // ETB Deviation outside limit detected, set timestamp and monitor it for two seconds
+        if(chVTIsSystemTimeWithinX(start_time, start_time + TIME_MS2I(2000))) {
+        	  //Less than two seconds, system OK: do nothing
+            }
+       else {
+         if(rpm > 600){  //We dont want or need any of this to execute if engine are not running
```

Comment:
600 is a magic number and I am suspisious of universal number of everyone. Maybe you want "2 * crankingRpm" or something like that?

Also https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad

Reply:
Yes, this one could be replaced by cranking rpm

Reply:
Don't directly use RPM, check that the current engine state is running

Reply:
if(rpm > engineConfiguration->cranking.rpm) ?